src/app-layer-parser.c

```diff
@@ -919,6 +918,23 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint32_t tofree_nb = 0;
+    if (total_txs > UINT32_MAX) {
+        SCFree(txl->tx_id_free_list);
```

Comment:
Why does `AppLayerParserGetTxCnt` return a u64 ?

Reply:
It's not named correctly, it should be something like `GetMaxId`

Reply:
Oh...

So tx count is bounded by MaxId - MinId...